File: mono/src/symbol/type_set.rs
use serde::{Serialize, Deserialize};

use crate::symbol::{Documentation, Declaration, TypeSignature};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Type {
    #[serde(flatten)]
    pub documentation: Documentation,

    /// Signature of the function
    pub r#type: String,

    /// Parsed function signature
    pub parsed_signature: Option<TypeSignature>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TypeSet {
    #[serde(flatten)]
    pub declaration: Declaration,

    /// Type signatures
    pub types: Vec<Type>,
}

File: typings/src/interfaces/symbol/type_set.ts
import { Declaration, Documentation } from './base';
import { TypeSignature } from './type_def';

/**
 * @note This extends Documentation instead of Declaration
 */
export interface Type extends Documentation {
    /**
     * @brief Signature of the function
     * @readonly
     */
    readonly type: string,

    /**
     * @brief Parsed function signature
     * @readonly
     */
    readonly parsedSignature: TypeSignature,
}

export interface TypeSet extends Declaration {
    /**
     * @brief Type signatures
     * @readonly
     */
    readonly types: Type[],
}
